libevmasm/Instruction.cpp

```diff
@@ -236,6 +238,8 @@ static std::map<Instruction, InstructionInfo> const c_instructionInfo =
 	{ Instruction::MSTORE8,		{ "MSTORE8",		0, 2, 0, true, Tier::VeryLow } },
 	{ Instruction::SLOAD,		{ "SLOAD",			0, 1, 1, false, Tier::Special } },
 	{ Instruction::SSTORE,		{ "SSTORE",			0, 2, 0, true, Tier::Special } },
+	{ Instruction::TLOAD,       { "TLOAD",          0, 1, 1, false, Tier::WarmAccess} },
+	{ Instruction::TSTORE,      { "TSTORE",         0, 2, 0, true, Tier::WarmAccess} },
```

Comment:
Yes, we should basically treat both of them as "worst side effects" like an external call or such for the time being.

Reply:
``SideEffectsCollector::movableRelativeTo()`` looks a bit dangerously implemented right now :-) - although nobody would have anticipated just introducing a new data region like this...

Reply:
Actually, I don't see how ``movableRelativeTo`` can be properly implemented without adding transient storage to ``SideEffects``.

Reply:
Even in the loop invariant code motion, this may very well already lead to moving a ``tload`` past a ``tstore`` in the loop, won't it? (Without having really looked into it)

Reply:
Good that we still found it then :-).

Reply:
I overlooked that part of semantic information when starting to work on this. I have covered the case @cameel pointed. I am still looking the code for more places where we might need to add coverage.